WeatherForecastApp/Controller/LocationDetailViewController.swift

```diff
@@ -120,16 +122,62 @@ extension LocationDetailViewController: UITableViewDelegate, UITableViewDataSour
     }
 }
 
-//extension LocationDetailViewController: UICollectionViewDataSource, UICollectionViewDelegate {
-//
-//    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
-//        return weatherDetail.dailyWeatherData.count
-//    }
-//
-//    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
-//        let hourlyCell = collectionView.dequeueReusableCell(withReuseIdentifier: "HourlyCell", for: indexPath)
-//        return hourlyCell
-//    }
-//
-//
-//}
+extension LocationDetailViewController: CLLocationManagerDelegate {
+    
+    func getLocation() {
+        locationManager = CLLocationManager()
+        locationManager.delegate = self
+    }
+    
+    func locationManager(_ manager: CLLocationManager, didChangeAuthorization status: CLAuthorizationStatus){
+        print("checking authentication status")
+        handleAuthorizationStatus(status: status)
+    }
+    
+    func handleAuthorizationStatus(status: CLAuthorizationStatus) {
+        switch status {
+            
+        case .notDetermined:
+            locationManager.requestWhenInUseAuthorization()
+        case .restricted:
+            self.oneButtonAlert(title: "location service denied", message: "parental control resricting location use")
+        case .denied:
+            break
```

Comment:
You need to handle this case, like what does your app do in this case because you require location to show weather